### PR TITLE
feature: support units for independent transforms

### LINKED ISSUE #111: [Feature] Support units for independent transforms
**1. Describe the bug**

In safari and firefox, an 'x' transform that uses a '%' unit is interpreted as 'px'

**2. IMPORTANT: Provide a CodeSandbox reproduction of the bug**

https://codesandbox.io/s/sleepy-rgb-1vdy77

**3. Steps to reproduce**

1. Go to the sandbox with Chrome
2. See how the 'x' animation is set to '100%'
3. See how it animates to the right 300px (100% of the element's width)
4. Go to the sandbox with Safari or Firefox
5. See how it animates to the right 100px instead

**4. Expected behavior**

It should animate to 100% of the with (300px) in all cases

Seeing the code, I saw that in this line

https://github.com/motiondivision/motionone/blob/c66431e49eeeba216a2c13550f0605e509b976ef/packages/dom/src/animate/utils/transforms.ts#L33

every transform has a default unit of 'px'.
I think such default unit should depend on the original unit stablished in the keyframe.

### PR DESCRIPTION
**Pull request by xaviergonz**
Fixes #111

### CODE REVIEW

File: packages/dom/src/animate/animate-style.ts [hunk 1]
```diff
@@ -278,3 +291,32 @@ export function animateStyle(
     return animation
   }
 }
+
+function extractKeyframeUnit(keyframe: string | number): string | undefined {
+  if (typeof keyframe !== "string") {
+    return undefined
+  }
+
+  let numberCharFound = false
```
Comment (MrHeer) [inline packages/dom/src/animate/animate-style.ts:300]: why `numberCharFound` never change? What it does?

File: packages/dom/src/animate/animate-style.ts [hunk 2]
```diff
@@ -220,6 +220,12 @@ export function animateStyle(
        * polyfill for transforms.
        */
     } else if (valueIsTransform) {
+      // We use the last keyframe to detect the unit since the first ones might be null.
+      const detectedUnit: string | undefined =
```
Comment (MrHeer) [inline packages/dom/src/animate/animate-style.ts:224]: `detectedUnit` just used in `render`, but why are they so far away?

**Comment by mattgperry** [comment]
Hey @xaviergonz thanks for the PR - I've used this test for https://github.com/motiondivision/motionone/pull/128 but adding support to timeline as well

File: packages/dom/src/animate/animate-style.ts (hunk 1 above)
Comment (xaviergonz) [reply packages/dom/src/animate/animate-style.ts:300]: good point, it was a leftover from a refactor. removed!

**Pull request closed without merging**